ProcessLib/HydroMechanics/HydroMechanicsFEM-impl.h

```diff
@@ -764,6 +764,35 @@ void HydroMechanicsLocalAssembler<ShapeFunctionDisplacement,
         ShapeFunctionPressure, typename ShapeFunctionDisplacement::MeshElement,
         DisplacementDim>(_element, _is_axially_symmetric, p,
                          *_process_data.pressure_interpolated);
+
+    auto sigma_eff_sum = MathLib::KelvinVector::tensorToKelvin<DisplacementDim>(
+        Eigen::Matrix<double, 3, 3>::Zero());
+
+    for (auto const& ip_data : _ip_data)
+    {
+        sigma_eff_sum += ip_data.sigma_eff;
+    }
+
+    Eigen::Matrix<double, 3, 3, 0, 3, 3> const sigma_avg =
+        MathLib::KelvinVector::kelvinVectorToTensor(sigma_eff_sum) /
+        _integration_method.getNumberOfPoints();
+
+    Eigen::SelfAdjointEigenSolver<Eigen::Matrix<double, 3, 3>> e_s(sigma_avg);
+
+    int const elem_id = _element.getID();
+
+    Eigen::Map<Eigen::Vector3d>(
+        &(*_process_data.principal_stress_values)[elem_id * 3], 3) =
+        e_s.eigenvalues();
+
+    auto eigen_vectors = e_s.eigenvectors();
+
+    for (auto i = 0; i < 3; i++)
+    {
+        Eigen::Map<Eigen::Vector3d>(
+            &(*_process_data.principal_stress_vector[i])[elem_id * 3], 3) =
+            eigen_vectors.col(i);
+    }
```

Comment:
Are the entries of eigen_vectors sorted?  σ_1 = max ( λ 1 , λ 2 , λ 3 ), σ 3 = min ( λ 1 , λ 2 , λ 3 ).

Reply:
According to the Eigen docu the eigenvalues are sorted in increasing order and the eigenvectors correspond to the eigenvalue of the same number. So it would be opposite: σ_1 = min ( λ 1 , λ 2 , λ 3 ), σ 3 = max ( λ 1 , λ 2 , λ 3 )